### PR TITLE
We've done(Nurgazy, Sunat) sing in, sing up

### PR DESCRIPTION
**Pull request by nurgazydev**

### CODE REVIEW

File: src/main/java/kg/peaksoft/giftlistb6/apies/UserApi.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import kg.peaksoft.giftlistb6.services.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/auth")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/apies/UserApi.java]: api/public

Comment (ChyngyzIshen) [reply src/main/java/kg/peaksoft/giftlistb6/apies/UserApi.java:15]: add cross origin

File: src/main/java/kg/peaksoft/giftlistb6/apies/UserApi.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+public class UserApi {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/apies/UserApi.java:14]: add cross origin

File: src/main/java/kg/peaksoft/giftlistb6/configs/security/TokenVerifyFilter.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package kg.peaksoft.giftlistb6.configs.security;
+import kg.peaksoft.giftlistb6.entities.User;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/configs/security/TokenVerifyFilter.java:3]: Write in the next line

File: src/main/java/kg/peaksoft/giftlistb6/configs/security/WebAppSecurity.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+public class WebAppSecurity {
+
+    @Bean
+    @SneakyThrows
+    SecurityFilterChain securityFilterChain(HttpSecurity httpSecurity, TokenVerifyFilter filter) {
+
+        httpSecurity.cors().and().csrf().disable()
+                .authorizeHttpRequests(auth -> {
+                    auth.antMatchers("/api-docs", "/v3/api-docs")
+                            .permitAll()
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/configs/security/WebAppSecurity.java:28]: add api/public/**

File: src/main/java/kg/peaksoft/giftlistb6/services/UserService.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepo;
+
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/services/UserService.java]: Unnecessary line

File: src/main/java/kg/peaksoft/giftlistb6/services/UserService.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    private final JwtUtils jwtUtils;
+
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/services/UserService.java]: Unnecessary line

**Review by i-Coderr — COMMENTED** [review]
Commented 12.10.22 at 3.45 pm

File: src/main/java/kg/peaksoft/giftlistb6/services/UserService.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import kg.peaksoft.giftlistb6.entities.User;
+import kg.peaksoft.giftlistb6.enums.Role;
+import kg.peaksoft.giftlistb6.exseptions.NotFoundException;
+import kg.peaksoft.giftlistb6.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/services/UserService.java]: Remove

File: src/main/java/kg/peaksoft/giftlistb6/services/UserService.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.List;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/services/UserService.java]: Remove

File: src/main/java/kg/peaksoft/giftlistb6/services/UserService.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    public User convertToRegisterEntity(RegisterRequest registerRequest)  {
+        return User.builder()
+                .firstName(registerRequest.getFirstName())
+                .lastName(registerRequest.getLastName())
+                .email(registerRequest.getEmail())
+                .password(registerRequest.getPassword())
+                .build();
+    }
+
+    public AuthResponse convertToView(User user) {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/services/UserService.java]: Method is never used

File: src/main/java/kg/peaksoft/giftlistb6/services/UserService.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+        return AuthResponse.builder()
+                .id(user.getId())
+                .email(user.getEmail())
+                .role(user.getRole())
+                .build();
+    }
+
+    public List<AuthResponse> convertEntitiesToView(List<User> users) {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/services/UserService.java]: Method is never used

File: src/main/java/kg/peaksoft/giftlistb6/services/UserService.java [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepo;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/services/UserService.java:21]: Write in the next line

File: src/main/java/kg/peaksoft/giftlistb6/services/UserService.java [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+    private final JwtUtils jwtUtils;
+    private final PasswordEncoder passwordEncoder;
+
+
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/services/UserService.java:25]: Unnecessary line

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/AuthResponse.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+@Setter
+@AllArgsConstructor
+@Builder
+public class AuthResponse {
+
+    private Long id;
+    private String email;
+    private Role role;
+    private String jwt;
+
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/AuthResponse.java:21]: We should send a success or error message to the front end

**Comment by i-Coderr** [comment]
Your code is not working, check properly before sending to the pull request

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/AuthResponse.java (hunk 13 above)
Comment (nurgazydev) [reply src/main/java/kg/peaksoft/giftlistb6/dto/responses/AuthResponse.java:21]: We have error messages if client enter an incorrect password or email(Global Exception Handler)
but why we should send a success message, if fronted understand successful saving user, looking at response (jwt role and email)

**Comment by nurgazydev** [comment]
I cheked before sending to the pull request, code was warking in my intellij and working now too

**Pull request merged**